CuriousKatie/ViewController.swift

```diff
@@ -14,19 +14,36 @@ class ViewController: UIViewController {
         super.viewDidLoad()
         
         /// Creates a list of random people.
-        let listOfParticipants = Helper.pickSomePeople()
+        let participants = Helper.pickSomePeople()
         
         /// Participants introduce themselves
-        for person in listOfParticipants {
+        for person in participants {
             print(person.introduceMyself())
         }
         
         /// Print a new line to separate sections.
         print("\n")
         
         /// All participants share all of their interests.
-        for person in listOfParticipants {
-            print(person.shareMyInterests())
+        for person in participants {
+            print(person.shareAllInterests())
+        }
+        
+        /// Print a new line to separate sections.
+        print("\n")
+        
+        /// Each participant shares one of their interests at a time.
+        var participantsWithInterests = participants.shuffled()
+        while participantsWithInterests.count != 0 {
+            participantsWithInterests = participantsWithInterests.shuffled()
+            for (index, person) in participantsWithInterests.enumerated() {
+                if let somethingToShare = person.shareNextInterest() {
+                    print(somethingToShare)
+                } else {
+                    participantsWithInterests.remove(at: index)
+                    break
```

Comment:
Alternatively (meaning FYI), you can remove the object without passing by the index if the object implements `Equatable`.
With this implementation, the `break` would become removable.